### PR TITLE
Infra, Docs: Publish Apache Iceberg 1.6.0 release

### PR DESCRIPTION
**Pull request by jbonofre**

### CODE REVIEW

**Comment by jbonofre** [comment]
@Fokko do you mind to take a look on this one ?

**Review by Fokko — APPROVED** [review]
Looking good 👍

**Comment by nastra** [comment]
@jbonofre can you please also update RevAPI to compare against the new version?

File: site/docs/releases.md [hunk 1]
```diff
@@ -71,6 +71,35 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
...
+    - Move to goooler shadow plugin ([\#10568](https://github.com/apache/iceberg/pull/10568))
+* Core
+    - Fix NPE during conflict handling of NULL partitions ([\#10680](https://github.com/apache/iceberg/pull/10680))
+    - Fix ParallelIterable memory leak where queue continues to be populated even after iterator close ([\#9402](https://github.com/apache/iceberg/pull/9402))
+    - REST Catalog: deprecate oauth/tokens endpoint ([\#10603](https://github.com/apache/iceberg/pull/10603))
+    - REST Catalog: fix create v1 table ([\#10369](https://github.com/apache/iceberg/pull/10369))
+    - REST Catalog: disallow overriding "credential" in table sessions ([\#10345](https://github.com/apache/iceberg/pull/10345))
+    - REST Catalog: assume issued_token_type is access_token to fully comply with RFC 6749 )[\#10314](https://github.com/apache/iceberg/pull/10314))
+    - JDBC Catalog: exclude unexpected namespaces ([\#10498](https://github.com/apache/iceberg/pull/10498))
+* Spark
```
Comment (nastra) [inline site/docs/releases.md:91]: given that there aren't many things listed here it gives the impression that it's a small release. I would probably just add a buch of more highlights from each component from https://github.com/apache/iceberg/releases/tag/apache-iceberg-1.6.0 (e.g. Flink is completely missing)

Comment (ajantha-bhat) [reply site/docs/releases.md:91]: +1, 
we both gave same comment at same time.

Comment (jbonofre) [reply site/docs/releases.md:91]: Yeah, good point, I missed the flink ones. Let me update it.

File: site/docs/releases.md [hunk 2]
```diff
@@ -71,6 +71,35 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
...
+    - Writes/actions should only perform cleanup if failure is cleanable ([\#10373](https://github.com/apache/iceberg/pull/10373))
+    - Fix issue when partitioning by UUID ([\#8250](https://github.com/apache/iceberg/pull/8250))
+* Kafka Connect
+    - Add commit coordinator ([\#10351](https://github.com/apache/iceberg/pull/10351))
+* Azure
+    - Make AzureProperties with shared-key credentials serializable ([\#10045](https://github.com/apache/iceberg/pull/10045))
+
+For more details, please visit [Github](https://github.com/apache/iceberg/releases/tag/apache-iceberg-1.6.0).
```
[the remark this answers is not in the document]
Comment (jbonofre) [reply site/docs/releases.md:99]: Gonna tackle that.

Generally speaking, I think the releases page should be a bit revisited and automatic, else it's not easy to update.

**Comment by ajantha-bhat** [comment]
@nastra, @jbonofre: Since release notes gonna take sometime, I have opened https://github.com/apache/iceberg/pull/10754 because I think open PR is failing (https://github.com/apache/iceberg/pull/10753#discussion_r1688048052) because we have a new tag and old version is not matching?

File: site/docs/releases.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jbonofre) [reply site/docs/releases.md:99]: I checked and multi-engine-support doc is already up to date.

[the remark this answers is not in the document]
Comment (ajantha-bhat) [reply site/docs/releases.md:99]: > I checked and multi-engine-support doc is already up to date.

Ack.

[the remark this answers is not in the document]
Comment (ajantha-bhat) [reply site/docs/releases.md:99]: I usually used to prepare release notes for non-patch release from community sync highlights since previous release. Thats how I did for 1.5.0, It covers most of the major things. I know it is quite a work. But without this it is hard to find the highlights of the release in the pool of few hundred/thousand commits for the user.

File: site/docs/releases.md (hunk 1 above)
Comment (jbonofre) [reply site/docs/releases.md:116]: I updated the highlights mixing community sync notes and full changelog.

File: site/docs/releases.md [hunk 3]
```diff
@@ -71,6 +71,90 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
...
 </dependencies>
 ```
+
+### 1.6.0 release
+Apache Iceberg 1.6.0 was released on July 23, 2024.
+
+The 1.6.0 release contains fixes and dependency updates.
+
+* API
+    - Fix default FileIO#newInputFile ManifestFile, DataFile and DeleteFile implementation to pass lengths ([\#9953](https://github.com/apache/iceberg/pull/9953))
```
Comment (ajantha-bhat) [inline site/docs/releases.md:81]: This fix is already released in 1.5.1, no need to mention for 1.6.0
https://github.com/apache/iceberg/pull/10114

Comment (jbonofre) [reply site/docs/releases.md:81]: It appears in the change log anyway. That's why I put there for consistency. I can remove it but I think it could be helpful for users.

Comment (ajantha-bhat) [reply site/docs/releases.md]: It was fixed in the main branch and then back ported to 1.5.x release. So, it appears in the change log. 

No need to mention it as it is not specific to this release.

File: site/docs/releases.md [hunk 4]
```diff
@@ -71,6 +71,88 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
   ...
 </dependencies>
 ```
+
+### 1.6.0 release
+Apache Iceberg 1.6.0 was released on July 23, 2024.
+
+The 1.6.0 release contains fixes and dependency updates.
```
Comment (Fokko) [inline site/docs/releases.md:78]: Also some features, right? Like the new Kafka connect etc?

**Review by Fokko — APPROVED** [review]
Looks good, thanks @jbonofre for working on this 👍 

@ajantha-bhat Wanted to check if you have any more comments before merging this

File: site/docs/releases.md (hunk 4 above)
Comment (jbonofre) [reply site/docs/releases.md:78]: That's right, let me rephrase.

**Review by nastra — CHANGES_REQUESTED** [review]
RevAPI update is still missing, see also my earlier comment: https://github.com/apache/iceberg/pull/10752#issuecomment-2245143821

**Comment by jbonofre** [comment]
@nastra revapi has been updated already by another PR, so not included in this one.

File: site/docs/releases.md [hunk 5]
```diff
@@ -71,6 +71,88 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
...
+    - support read of partition metadata column when table is over 1k ([\#10641](https://github.com/apache/iceberg/pull/10641))
+    - Fix the setting of equalAuthorities in RemoveOrphanFilesProcedure ([\#10342](https://github.com/apache/iceberg/pull/10342))
+    - Fix system function pushdown in CoW row-level commands ([\#10119](https://github.com/apache/iceberg/pull/10119))
+    - Only traverse ancestors of current snapshot when building changelog scan ([\#10405](https://github.com/apache/iceberg/pull/10405))
+    - Add max allowed failed commits to RewriteDataFiles when partial progress is enabled ([\#9611](https://github.com/apache/iceberg/pull/9611))
+    - Fix issue when partitioning by UUID ([\#8250](https://github.com/apache/iceberg/pull/8250))
+    - Use bulk deletes in rewrite manifests action ([\#10343](https://github.com/apache/iceberg/pull/10343))
+* Flink
+    - Remove Flink 1.16 support ([\#10154](https://github.com/apache/iceberg/pull/10154))
+    - Adds support for Flink 1.19 version ([\#10112](https://github.com/apache/iceberg/pull/10112))
```
Comment (nastra) [inline site/docs/releases.md:127]:
```suggestion
    - Add support for Flink 1.19 ([\#10112](https://github.com/apache/iceberg/pull/10112))
```

File: site/docs/releases.md [hunk 6]
```diff
@@ -71,6 +71,88 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
...
+* Core
+    - Add EnvironmentContext to commit summary ([\#9273](https://github.com/apache/iceberg/pull/9273))
+    - Add explicit JSON parser for ConfigResponse ([\#9952](https://github.com/apache/iceberg/pull/9952))
+    - Calling rewrite_position_delete_files fails on tables with more than 1k columns ([\#10020](https://github.com/apache/iceberg/pull/10020))
+    - Expose table incremental scan for appends API in SerializableTable ([\#10682](Expose table incremental scan for appends API in SerializableTable))
+    - Fix NPE during conflict handling of NULL partitions ([\#10680](https://github.com/apache/iceberg/pull/10680))
+    - Fix ParallelIterable memory leak where queue continues to be populated even after iterator close ([\#9402](https://github.com/apache/iceberg/pull/9402))
+    - Fix logging table name in scanning metadata table ([\#10141](https://github.com/apache/iceberg/pull/10141))
+    - Lazily compute & cache hashCode in CharSequenceWrapper ([\#10023](https://github.com/apache/iceberg/pull/10023))
+    - pushdown data_file.content when filter manifests in entries table ([\#10203](https://github.com/apache/iceberg/pull/10203))
```
Comment (nastra) [inline site/docs/releases.md:91]:
```suggestion
    - Pushdown data_file.content when filter manifests in entries table ([\#10203](https://github.com/apache/iceberg/pull/10203))
```

File: site/docs/releases.md [hunk 7]
```diff
@@ -71,6 +71,88 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
...
+    - use bulk delete when removing old metadata.json files ([\#10679](https://github.com/apache/iceberg/pull/10679))
```
Comment (nastra) [inline site/docs/releases.md:92]:
```suggestion
    - Use bulk delete when removing old metadata.json files ([\#10679](https://github.com/apache/iceberg/pull/10679))
```

File: site/docs/releases.md [hunk 8]
```diff
@@ -71,6 +71,88 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
...
+    - Record converters ([\#9641](https://github.com/apache/iceberg/pull/9641))
+    - Handle namespace creation for auto table creation ([\#10186](https://github.com/apache/iceberg/pull/10186))
+    - Update iceberg.hadoop-conf-dir config description ([\#10184](https://github.com/apache/iceberg/pull/10184))
+* Parquet
+    - Don't write column sizes when metrics mode is None ([\#10440](https://github.com/apache/iceberg/pull/10440))
+* Spark
+    - Fix handling of null binary values when sorting with zorder ([\#10026](https://github.com/apache/iceberg/pull/10026))
+    - Spark writes/actions should only perform cleanup if failure is cleanable ([\#10373](https://github.com/apache/iceberg/pull/10373))
+    - Use 'delete' if RowDelta only has delete files ([\#10123](https://github.com/apache/iceberg/pull/10123))
+    - support read of partition metadata column when table is over 1k ([\#10641](https://github.com/apache/iceberg/pull/10641))
```
Comment (nastra) [inline site/docs/releases.md:118]:
```suggestion
    - Support read of partition metadata column when table is over 1k ([\#10641](https://github.com/apache/iceberg/pull/10641))
```

File: site/docs/releases.md [hunk 9]
```diff
@@ -71,6 +71,88 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
...
+    - OpenAPI: TableRequirements should use union of subclasses ([\#10434](https://github.com/apache/iceberg/pull/10434))
+    - OpenAPI: Deprecate oauth/tokens endpoint ([\#10603](https://github.com/apache/iceberg/pull/10603))
+    - OpenAPI: Fix additionalProperties for SnapshotSummary ([\#9838](https://github.com/apache/iceberg/pull/9838))
+    - OpenAPI: Fix property names for stats/partition stats ([\#10662](https://github.com/apache/iceberg/pull/10662))
+* Vendor Integrations
+    - AWS: Make sure Signer + User Agent config are both applied ([\#10198](https://github.com/apache/iceberg/pull/10198))
+    - AWS: Retain Glue Catalog column comment after updating Iceberg table ([\#10276](https://github.com/apache/iceberg/pull/10276))
+    - AWS: Retain Glue Catalog table description after updating Iceberg table ([\#10199](https://github.com/apache/iceberg/pull/10199))
+    - AWS: Support S3 DSSE-KMS encryption ([\#8370](https://github.com/apache/iceberg/pull/8370))
+    - AWS: close underlying executor for DynamoDb LockManager ([\#10132](https://github.com/apache/iceberg/pull/10132))
```
Comment (nastra) [inline site/docs/releases.md:148]:
```suggestion
    - AWS: Close underlying executor for DynamoDb LockManager ([\#10132](https://github.com/apache/iceberg/pull/10132))
```

**Comment by nastra** [comment]
> @nastra revapi has been updated already by another PR, so not included in this one.

ah sorry I missed that update

**Review by nastra — APPROVED** [review]
LGTM with a few minor comments

File: site/docs/releases.md [hunk 10]
```diff
@@ -71,6 +71,88 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
...
+    - OpenAPI: Fix property names for stats/partition stats ([\#10662](https://github.com/apache/iceberg/pull/10662))
+* Vendor Integrations
+    - AWS: Make sure Signer + User Agent config are both applied ([\#10198](https://github.com/apache/iceberg/pull/10198))
+    - AWS: Retain Glue Catalog column comment after updating Iceberg table ([\#10276](https://github.com/apache/iceberg/pull/10276))
+    - AWS: Retain Glue Catalog table description after updating Iceberg table ([\#10199](https://github.com/apache/iceberg/pull/10199))
+    - AWS: Support S3 DSSE-KMS encryption ([\#8370](https://github.com/apache/iceberg/pull/8370))
+    - AWS: Close underlying executor for DynamoDb LockManager ([\#10132](https://github.com/apache/iceberg/pull/10132))
+    - AWS: Add Iceberg version to UserAgent in S3 requests ([\#9963](https://github.com/apache/iceberg/pull/9963))
+    - Azure: Make AzureProperties w/ shared-key creds serializable ([\#10045](https://github.com/apache/iceberg/pull/10045))
+
```
Comment (nastra) [inline site/docs/releases.md:151]: should this also have a `Dependencies` section like 1.5.0? People pointed out in the past that it's quite helpful to immediately see which dependencies have been updated

Comment (jbonofre) [reply site/docs/releases.md:151]: Good point. I thought dependabot updates are not super helpful, but if people said they are, I'm happy to add this.
I'm updating the PR.

**Review by ajantha-bhat — APPROVED** [review]
LGTM. 

Dependency update is good to have along with this.

**Review by hussein-awala — APPROVED** [review]

**Review by amogh-jahagirdar — APPROVED** [review]
Thanks @jbonofre! Thanks @ajantha-bhat @Fokko @nastra @hussein-awala for reviewing.

File: site/docs/releases.md (hunk 10 above)
Comment (jbonofre) [reply site/docs/releases.md:151]: Sorry, I forgot to update this section yesterday. I'm doing it now.

Comment (jbonofre) [reply site/docs/releases.md:163]: I added `Dependencies` section for key updates.

**Comment by jbonofre** [comment]
@amogh-jahagirdar as requested by @nastra I added a `Dependencies` section

**Comment by amogh-jahagirdar** [comment]
Thanks @jbonofre !

**Comment by jbonofre** [comment]
@amogh-jahagirdar thanks ! I will do the announcement.
